client/src/components/forms/select/autocomplete/types.d.ts

```diff
@@ -16,4 +16,5 @@ export type AutoCompleteSelectProps<T = string> = Omit<
   rowHeight?: number;
   clearable?: boolean;
   onClearSelection?: () => void;
+  onSearch?: (value: string) => void;
```

Comment:
`(value: T) => void;` better, so we can ensure the returned value is of the same type as we set in the options.

Reply:
Anyway, why do you need this `onSearch` callback for? 🤔